src/main/java/org/skriptlang/skript/bukkit/spark/SparkUtils.java

```diff
@@ -0,0 +1,162 @@
+package org.skriptlang.skript.bukkit.spark;
+
+import me.lucko.spark.api.Spark;
+import me.lucko.spark.api.SparkProvider;
+import me.lucko.spark.api.statistic.StatisticWindow.CpuUsage;
+import me.lucko.spark.api.statistic.StatisticWindow.MillisPerTick;
+import me.lucko.spark.api.statistic.StatisticWindow.TicksPerSecond;
+import me.lucko.spark.api.statistic.misc.DoubleAverageInfo;
+import me.lucko.spark.api.statistic.types.DoubleStatistic;
+import me.lucko.spark.api.statistic.types.GenericStatistic;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.UnknownNullability;
+
+import java.util.Arrays;
+
+/**
+ * Utility class for accessing Spark's API.
+ */
+public class SparkUtils {
```

Comment:
I think this class would benefit from having a better lifecycle, mainly to orchestrate its state a bit better.
What I recommend is adding and refactoring the following:
- rename to something else. This is hardly a utility class, it's a stateful singleton.
- Add an initialize() method that consumes the desired Statistic classes (process cpu usage, system cpu usage, mspt, tps)
- Add a dispose() method that de-initializes the state and prepares this singleton to be re-initialized at another time
- Manage this lifecycle from somewhere, probably SparkModule. Ultimately, onEnable should call initialize() and onDisable should call dispose()

This will clean up pretty much all of your synchronized blocks, which barely needed to be there in the first place.

Reply:
I don't even know how to do all of that, but it works fine as it is, so I will leave it.